src/components/moviesSection/moviesSection.tsx

```diff
@@ -10,19 +13,35 @@ import styles from './moviesSection.module.css'
 
 export const MoviesSection: React.FC = () => {
   const [moviesData, setMoviesData] = React.useState(movies)
+  const [selectedMovieName, setSelectedMovieName] = React.useState('')
+  const [selectedMovieYear, setSelectedMovieYear] = React.useState(0)
+
+  const { isDisplayed, toggleModal } = useModal()
 
   const handleSort = () => {
     const moviesDataCopy = [...moviesData]
     moviesDataCopy.sort(sortByYear)
     setMoviesData(moviesDataCopy)
   }
```

Comment:
you can use useCallback here

```suggestion
  const handleSort = React.useCallback(() => {
    const moviesDataCopy = [...moviesData]
    moviesDataCopy.sort(sortByYear)
    setMoviesData(moviesDataCopy)
  }, [moviesData])
```